prow/crier/reporters/gerrit/reporter.go

```diff
@@ -622,74 +715,6 @@ func GenerateReport(pjs []*v1.ProwJob, customCommentSizeLimit int) JobReport {
 		// We don't care about other states, so keep original order.
 		return true
 	})
-
-	// TODO(listx): Clean this up (whether we include or exclude URLs). We
-	// should probably just construct an optimistic report (with full URLs), and
-	// then decide to trim it down if we are over the commentSizeLimit.
-	//
-	// Another thing we can do is do text-to-text compression so that we
-	// (almost) never skip reporting. E.g., see
-	// https://stackoverflow.com/a/41188719/437583. This should suffice for most
-	// scenarios because most of the time the job URLs share a large prefix
-	// (ideal for compression).
-	//
-	// Additionally, note that newline characters are very special here because
-	// they are used as token delimiters during deserialization (see
-	// https://github.com/kubernetes/test-infra/blob/b45b20a405a82de65d56196da00f6106b841dd40/prow/gerrit/adapter/adapter.go#L260).
-	// The use of newline characters is most likely a result of Gerrit comments
-	// only supporting plaintext (and thus, needing to use a delimiter that is
-	// not awkward on human eyes).
-	remainingSize := commentSizeLimit - len(fullHeader(report.Header, reTestMessage))
-	linesWithURLs := make([]string, len(report.Jobs))
-	linesWithoutURLs := make([]string, len(report.Jobs))
-	for i, job := range report.Jobs {
-		// If the URL cannot be found (for example, because the job has not been
-		// scheduled due to some other failure), then use the string
-		// URL_NOT_FOUND in its place instead. This way, we don't report jobs
-		// with an empty URL field.
-		if job.URL == "" {
-			job.URL = urlNotFound
-		}
-		linesWithURLs[i] = job.serialize()
-		remainingSize -= len(linesWithURLs[i])
-	}
-
-	// cutoff is the index where if it's the line contains URL it will exceed the
-	// size limit.
-	cutoff := len(report.Jobs)
-	for cutoff > 0 && remainingSize < 0 {
-		cutoff--
-		linesWithoutURLs[cutoff] = report.Jobs[cutoff].serializeWithoutURL()
-		// remainingSize >= 0 after this condition means next line is not good
-		// to include URL
-		remainingSize += len(linesWithURLs[cutoff]) - len(linesWithoutURLs[cutoff])
-	}
-
-	// This shouldn't happen unless there are too many prow jobs(e.g. > 300) and
-	// each job name is super long(e.g. > 50)
-	if remainingSize < 0 {
-		report.Header = fullHeader(report.Header, " 👉 Comment '/test all' to rerun all tests")
-		report.Message = errorMessageLine("Prow failed to report all jobs, are there excessive amount of prow jobs?")
```

Comment:
when this happened, there might be jobs that failed and not presented in message, since `/retest` only retest what's available from previous prow comments, these jobs that were not included in report would not be triggered either.

Note that this shouldn't happen as we can simple say that we don't support >300 presubmit jobs, so 🤷

Reply:
To summarize, if there are too many jobs (and we hit the comment limit):

1. display the error message only (do not report any jobs, thus avoiding "polluting" `/retest`; old behavior)
2. display as many jobs as possible and also an error message (new behavior in this PR).

I think reporting something is better than nothing. Also if there are just too many jobs, that's something the user should fix anyway. So, if there are too many jobs and `/retest` doesn't actually retest everything (polluted...), it's still OK because at least the user can still interact with Prow to test some jobs that can be reported.

Reply:
Yeah I think it'll be fine, as prow will not vote on the label when only part of the failed jobs were retried